api/src/main/java/net/kyori/adventure/util/Index.java

```diff
@@ -175,6 +209,37 @@ private Index(final Map<K, V> keyToValue, final Map<V, K> valueToKey) {
     return this.keyToValue.get(key);
   }
 
+  /**
+   * Gets a value by its key.
+   *
+   * @param key the key
+   * @return the value
+   * @throws NoSuchElementException if there is no value for the key
+   * @since 4.11.0
+   */
+  public @NotNull V valueOrThrow(final @NotNull K key) {
+    final V value = this.value(key);
+    if (value == null) {
+      throw new NoSuchElementException("There is no value for key " + key);
+    }
+    return value;
+  }
+
+  /**
+   * Gets a value by its key or returns
+   * a fallback value.
+   *
+   * @param key the key
+   * @param defaultValue the fallback value
+   * @return the value
+   * @since 4.11.0
+   */
+  @Contract("_, null -> null; _, !null -> !null")
+  public V valueOrDefault(final @NotNull K key, @Nullable V defaultValue) {
```

Comment:
```suggestion
  public V valueOr(final @NotNull K key, @Nullable V defaultValue) {
```

to match `ComponentSerializer.serializeOr`